File: FinishProject/ChineseAuctionAngular/ChineseAuctionAngular/src/app/models/GiftDto.ts
export class Gift {
    idGift: number = 0;
    name: string = '';
    description?: string;
    categoryId: number = 0;
    category: any = null; 
    quantity: number = 1;
    price: number = 0;
    image?: string;
    idDonor: number = 0;
    donor: any = null;
    isDrawn: boolean = false;
    userId?: number;
    user?: any;
    giftOrders: any[] = [];

    constructor(init?: Partial<Gift>) {
        Object.assign(this, init);
    }
}

export class GiftDto {
    name: string = '';
    description?: string;
    categoryId: number = 0;
    quantity: number = 1;
    price: number = 0;
    image?: string;
    idDonor: number = 0;

    constructor(init?: Partial<GiftDto>) {
        Object.assign(this, init);
    }
}

export class GiftDtoNew {
    idGift: number = 0;
    name: string = '';
    description: string = '';
    price: number = 0;
    numOfBuyers: number = 0;

    constructor(init?: Partial<GiftDtoNew>) {
        Object.assign(this, init);
    }
}

File: ChineseAuction.Api.Angular/server/ChineseAuctionAPI/DTOs/GiftDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ChineseAuctionAPI.DTOs
{
    public class GiftDTO
    {
        public int IdGift { get; set; } // חובה לעדכון!

        [Required]
        public string Name { get; set; }

        [StringLength(500)]
        public string? Description { get; set; }

        [Required]
        public int CategoryId { get; set; }

        [Required, Range(1, 500)]
        public int Amount { get; set; } // שיניתי מ-Quantity ל-Amount שיתאים לאנגולר

        public string? Image { get; set; }

        [Required]
        public int IdDonor { get; set; }
    }
    public class GiftNewDTO
    {
        public int IdGift { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int NumOfBuyers { get; set; }
    }
    public class GiftUpdateDto
    {
        public int UserId { get; set; }
        public int GiftId { get; set; }
        public int Amount { get; set; }
    }
}
